src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ChimericAlignment.java

```diff
@@ -59,89 +56,231 @@
 
-        this.assemblyId = assemblyId;
-        this.contigId = contigId;
         this.regionWithLowerCoordOnContig = regionWithLowerCoordOnContig;
         this.regionWithHigherCoordOnContig = regionWithHigherCoordOnContig;
-        this.homology = homology;
-        this.insertedSequence = insertedSequence;
+
+        this.strandSwitch = determineStrandSwitch(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig);
+        this.involvesReferenceIntervalSwitch = involvesRefPositionSwitch(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, samSequenceDictionary);
+        this.isForwardStrandRepresentation = isForwardStrandRepresentation(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, this.strandSwitch, this.involvesReferenceIntervalSwitch);
+        this.isNotSimpleTranslocation = isNotSimpleTranslocation(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, this.strandSwitch, this.involvesReferenceIntervalSwitch);
+
+        this.contigSeq = contigSequence;
+
         this.insertionMappings = insertionMappings;
     }
 
-    @SuppressWarnings("unchecked")
-    protected ChimericAlignment(final Kryo kryo, final Input input) {
-        this.assemblyId = input.readString();
-        this.contigId = input.readString();
-        this.regionWithLowerCoordOnContig = kryo.readObject(input, AlignmentRegion.class);
-        this.regionWithHigherCoordOnContig = kryo.readObject(input, AlignmentRegion.class);
-        this.homology = input.readString();
-        this.insertedSequence = input.readString();
-        this.insertionMappings = (ArrayList<String>) kryo.readObject(input, ArrayList.class);
+    @VisibleForTesting
+    static StrandSwitch determineStrandSwitch(final AlignmentRegion first, final AlignmentRegion second) {
+        if (first.forwardStrand == second.forwardStrand) {
+            return StrandSwitch.NO_SWITCH;
+        } else {
+            return first.forwardStrand ? StrandSwitch.FORWARD_TO_REVERSE : StrandSwitch.REVERSE_TO_FORWARD;
+        }
     }
 
     /**
-     * TODO: never used/tested method, should remove?
-     *  Parses a tab-delimited assembled breakpoint line into an ChimericAlignment object. Fields should be in the same order as that produced by toString():
-     *
-     *  assemblyId
-     *  contigId
-     *  alignmentRegion1.toString()
-     *  alignmentRegion2.toString()
-     *  insertedSequence
-     *  homology
-     *
+     * Determine if the region that maps to a lower coordinate on the contig also maps to a lower coordinate on the reference.
+     */
+    @VisibleForTesting
+    static boolean involvesRefPositionSwitch(final AlignmentRegion regionWithLowerCoordOnContig,
+                                             final AlignmentRegion regionWithHigherCoordOnContig,
+                                             final SAMSequenceDictionary samSequenceDictionary) {
```

Comment:
The no translocations restriction is handled later (currently in the code, if you think we can enforce it here, we can do it, but once we want to handle simple translocations, we must use it in someway, either the dictionary of `SVIntervals`), so that assumption is not guaranteed here.